apis/v1alpha1/route_types.go

```diff
@@ -19,16 +19,39 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// GatewayObjectReference identifies a Gateway object.
-type GatewayObjectReference struct {
-	// Namespace is the namespace of the referent.
+// GatewayAllowType specifies which Gateways should be allowed to use a Route.
+type GatewayAllowType string
+
+const (
+	// GatewayAllowAll indicates that all Gateways will be able to use this
+	// route.
+	GatewayAllowAll GatewayAllowType = "All"
+	// GatewayAllowFromList indicates that only Gateways that have been
+	// specified in GatewayRefs will be able to use this route.
+	GatewayAllowFromList GatewayAllowType = "FromList"
+	// GatewayAllowSameNamespace indicates that only Gateways within the same
+	// namespace will be able to use this route.
+	GatewayAllowSameNamespace GatewayAllowType = "SameNamespace"
```

Comment:
nit: As noted in the pre-alpha feedback, we should organize these constants from least specific to most specific. We c an argue either ways on which of SameNamespace and FromList is most specific. Accept/reject this change as you see fit, no biggie.

```suggestion
	// GatewayAllowAll indicates that all Gateways will be able to use this
	// route.
	GatewayAllowAll GatewayAllowType = "All"
	// GatewayAllowSameNamespace indicates that only Gateways within the same
	// namespace will be able to use this route.
	GatewayAllowSameNamespace GatewayAllowType = "SameNamespace"
	// GatewayAllowFromList indicates that only Gateways that have been
	// specified in GatewayRefs will be able to use this route.
	GatewayAllowFromList GatewayAllowType = "FromList"
```

Reply:
This is a tough one, I'd initially thought of the `FromList` option as primarily being used to allow Gateways from other namespaces to use a Route, so I'd thought of it as less restrictive than SameNamespace. Definitely depends on the use case though. One tiny advantage of the current order is that it also happens to be alphabetical, not that that really matters.

Reply:
Let's skip this change for now.